### DISCUSSION TITLE
Would it make sense to support Span<byte>?

### DISCUSSION THREAD

**Discussion by kmgallahan** [dlemstra/Magick.NET#953] [General] [answered]
Hi. As the title indicates, would it make sense to support creation of `MagickImage` objects by passing in a `Span<byte>` to avoid allocating new `byte[]`'s / streams?

I'm not familiar with the Magick.NET internals, so I'm not sure how much of a pure wrapper it is.

**Comment by dlemstra** [comment, marked as answer]
It could be possible to add support for span but that would require me to either go from `netstandard20` to `netstandard21` or to add a dependency to be able to use `Span<T>` with  `netstandard20`. The latter sounds like a bad plan to me. But I am not sure if I want to switch to `2.1` at this point in time. This would break support for .NET Core 2.1.

And a third option would be also adding support for `netstandard21` besides supporting `netstandard20` but I really don't want to add another target platform.

**Reply by kmgallahan** [reply]
Would a constructor taking an IntPtr + length be too messy? Again, not sure if Magick.NET ever touches the data before passing it to native code. I'm just curious because I'm currently consuming an unmanaged image buffer and so copying it to a byte array just to have it turned back into a pointer seems wasteful.

**Reply by dlemstra** [reply]
I published a new release a week ago that adds new `Span<T>` overloads in various locations.

**Reply by kmgallahan** [reply]
Excellent, I'll take advantage of this asap.
